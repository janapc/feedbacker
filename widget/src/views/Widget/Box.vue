<template>
  <div class="box animate__animated animate_fadeInUp animate__faster" id="box">
    <header
      :class="{
        'justify-between': canShowAdditionalControlAndInfo,
        'justify-end': !canShowAdditionalControlAndInfo
      }"
      class="relative w-full flex"
    >
      <button
        id="btn-goBack"
        v-if="canShowAdditionalControlAndInfo"
        @click="back"
        :disabled="canGoBack"
        :class="{ invisible: canGoBack }"
        class="text-xl text-gray-800 focus:outline-none"
      >
        <icon name="ArrowRight" :color="colors.gray['800']" />
      </button>

      <p
        id="label"
        v-if="canShowAdditionalControlAndInfo"
        class="text-xl font-black text-center text-brand-main"
      >
        {{ label }}
      </p>

      <button
        id="close-box"
        @click="() => emit('close-box')"
        class="text-xl text-gray-800 focus:outline-none"
      >
        <icon size="14" name="Close" :color="colors.gray['800']" />
      </button>
    </header>

    <wizard />

    <footer
      class="text-gray-800 text-sm flex"
      v-if="canShowAdditionalControlAndInfo"
    >
      <icon class="mr-1" name="Chat" :color="brandColors.graydark" />
      widget by
      <span class="ml-1 font-bold">feedbacker</span>
    </footer>
  </div>
</template>

<script lang="ts">
import { defineComponent, computed, ComputedRef, SetupContext } from 'vue';
import colors from 'tailwindcss/colors';
import { brand } from '../../../palette';

import useStore from '../../hooks/useStore';
import useNavigation, { UseNavigation } from '../../hooks/useNavigation';

import Icon from '../../components/Icon/index.vue';
import Wizard from '../../components/Wizard/index.vue';

interface SetupReturn {
  emit: SetupContext['emit'];
  label: ComputedRef<string>;
  canGoBack: ComputedRef<boolean>;
  canShowAdditionalControlAndInfo: ComputedRef<boolean>;
  brandColors: Record<string, string>;
  colors: Record<string, string>;
  back: UseNavigation['back'];
}

export default defineComponent({
  emits: ['close-box'],
  components: { Icon, Wizard },
  setup (_, { emit }: SetupContext): SetupReturn {
    const store = useStore();
    const { back } = useNavigation();

    const label = computed<string>(() => {
      if (store.feedbackType === 'ISSUE') {
        return 'Reporte um problema';
      }
      if (store.feedbackType === 'IDEA') {
        return 'Nos false a sua idéia';
      }

      if (store.feedbackType === 'OTHER') {
        return 'Abra sua mente';
      }
      return 'O que você tem em mente?';
    });
    const canGoBack = computed<boolean>(() => {
      return store.currentComponent === 'SelectFeedbackType';
    });

    const canShowAdditionalControlAndInfo = computed<boolean>(() => {
      return (
        store.currentComponent !== 'Success' &&
        store.currentComponent !== 'Error'
      );
    });

    return {
      emit,
      label,
      canGoBack,
      canShowAdditionalControlAndInfo,
      colors,
      brandColors: brand,
      back
    };
  }
});
</script>

<style lang="postcss" scoped>
.box {
  @apply fixed z-50 bottom-0 right-0 mb-5 mr-5 bg-white rounded-xl py-5 px-5 flex flex-col items-center shadow-xl select-none;
  width: 400px;
}
</style>
