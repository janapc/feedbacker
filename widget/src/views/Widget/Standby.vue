<template>
  <div
    id="open-box"
    class="fixed z-50 bottom-0 right-0 mb-5 mr-5 bg-brand-main rounded-full py-3 px-5 flex items-center shadow-xl cursor-pointer select-none animate__animated animate__fadeInUp animate__faster"
    @click="() => emit('open-box')"
  >
    <icon name="Chat" color="white" size="27" class="mr-3" />
    <span class="font-black text-white text-xl">Deixe um feedback</span>
  </div>
</template>

<script lang="ts">
import { defineComponent, SetupContext } from 'vue';

import Icon from '../../components/Icon/index.vue';

interface SetupReturn {
  emit: SetupContext['emit'];
}

export default defineComponent({
  components: {
    Icon
  },
  emits: ['open-box'],
  setup (_, { emit }: SetupContext): SetupReturn {
    return { emit };
  }
});
</script>
