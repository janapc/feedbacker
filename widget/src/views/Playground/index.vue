<template>
  <div
    class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-main"
  >
    <h1 class="text-6xl font-black text-brand-gray">Playground</h1>
    <p class="text-2xl mt-3 font-regular text-brand-gray">
      Este Ã© o playground, use para testar o widget.
    </p>
  </div>

  <div
    class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray"
  >
    <h1 class="text-6xl font-black text-brand-graydark">ğŸš€</h1>
  </div>

  <div
    class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-main"
  >
    <h1 class="text-6xl font-black text-brand-gray">ğŸ‘€</h1>
  </div>

  <div
    class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray"
  >
    <h1 class="text-6xl font-black text-brand-gray">ğŸš¨</h1>
  </div>

  <widget />
</template>

<script lang="ts">
import { defineComponent } from 'vue';

import Widget from '../Widget/index.vue';

export default defineComponent({
  components: { Widget }
});
</script>
