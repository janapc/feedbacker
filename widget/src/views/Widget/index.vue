<template>
  <teleport to="body">
    <component
      @open-box="handleOpenBox"
      @close-box="handleCloseBox"
      :is="state.component"
    />
  </teleport>
</template>

<script lang="ts">
import { defineComponent, reactive, watch } from 'vue';

import useIframeControl from '../../hooks/useIframeControl';

import Standby from './Standby.vue';
import Box from './Box.vue';

import useStore from '../../hooks/useStore';

type State = {
  component: string;
};

interface SetupReturn {
  state: State;
  handleOpenBox(): void;
  handleCloseBox(): void;
}

export default defineComponent({
  components: { Standby, Box },
  setup (): SetupReturn {
    const iframeControl = useIframeControl();

    const state = reactive<State>({
      component: 'Standby'
    });

    const store = useStore();

    watch(
      () => store.currentComponent,
      () => {
        iframeControl.updateCoreValuesOnStore();
      }
    );

    function handleOpenBox (): void {
      iframeControl.notifyOpen();
      state.component = 'Box';
    }

    function handleCloseBox (): void {
      iframeControl.notifyClose();
      state.component = 'Standby';
    }

    return {
      state,
      handleOpenBox,
      handleCloseBox
    };
  }
});
</script>
