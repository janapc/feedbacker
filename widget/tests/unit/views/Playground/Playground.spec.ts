import { mount } from '@vue/test-utils';

import Playground from '../../../../src/views/Playground/index.vue';
import Widget from '../../../../src/views/Widget/index.vue';

describe('<Playground />', () => {
  it('Should render Playground', () => {
    const wrapper = mount(Playground);

    expect(wrapper.findAll('h1').length).toBe(4);

    expect(wrapper.text()).toContain('Playground');

    expect(wrapper.findComponent(Widget).exists()).toBeTruthy();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
