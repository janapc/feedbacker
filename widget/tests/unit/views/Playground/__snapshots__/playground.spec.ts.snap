// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playground /> Should render Playground 1`] = `
<div class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-main">
  <h1 class="text-6xl font-black text-brand-gray">Playground</h1>
  <p class="text-2xl mt-3 font-regular text-brand-gray"> Este Ã© o playground, use para testar o widget. </p>
</div>
<div class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray">
  <h1 class="text-6xl font-black text-brand-graydark">ğŸš€</h1>
</div>
<div class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-main">
  <h1 class="text-6xl font-black text-brand-gray">ğŸ‘€</h1>
</div>
<div class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray">
  <h1 class="text-6xl font-black text-brand-gray">ğŸš¨</h1>
</div>
<!--teleport start-->
<!--teleport end-->
`;
